aws/resource_aws_ec2_capacity_reservation_test.go

```diff
@@ -0,0 +1,222 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_ec2_capacity_reservation", &resource.Sweeper{
+		Name: "aws_ec2_capacity_reservation",
+		F:    testSweepEc2CapacityReservations,
+	})
+}
+
+func testSweepEc2CapacityReservations(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).ec2conn
+
+	resp, err := conn.DescribeCapacityReservations(&ec2.DescribeCapacityReservationsInput{})
+
+	if err != nil {
+		return fmt.Errorf("Error retrieving capacity reservations: %s", err)
+	}
+
+	if len(resp.CapacityReservations) == 0 {
+		log.Print("[DEBUG] No capacity reservations to sweep")
+		return nil
+	}
+
+	for _, r := range resp.CapacityReservations {
+		if *r.State != "cancelled" {
```

Comment:
This should also skip `expired` state capacity reservations. 😄 

```suggestion
		if aws.StringValue(r.State) != ec2.CapacityReservationStateCancelled && aws.StringValue(r.State) != ec2.CapacityReservationStateExpired {
```